### ISSUE TITLE
Endpoint for managing users

### ISSUE THREAD

**Issue by ryangrimm** [marklogic-community/Corona#62]
Almost every app has users.  Would be fantastic if Corona had endpoints to:

• Create a new user with a specific set of roles
• Authenticate users, getting back tokens for subsequent requests
• Username reminders with emails formatted using stored transformers
• Password reminders with emails formatted using stored transformers
• Change password
• Account removal endpoint
• Other handy stuff we think up along the way

**Comment by hunterhacker** [comment]
Parse.com has built-in support for users:
https://www.parse.com/docs/ios_guide#users

Takes a lot of burden away from the developer.  I think we have a lot to do before this becomes a priority though.

**Comment by collwhit** [comment]
Ryan, would you see a developer being able to do this?  Developer-admin?  Or just admin?

**Comment by ryangrimm** [comment]
I'm dreaming of support for end users of an application.  To follow that dream a user with the corona-dev role would need to be able to create a new user.  I can elaborate on why if need be.

**Comment by collwhit** [comment]
Ok!

What's your schedule the first week in January? I want to sit down with you, Charles and Erik and talk through some questions/ideas. 

If I arrange it, are you in?   I know Jason will be in Singapore. 

--Colleen

**Comment by ryangrimm** [comment]
Yes, I'll be around the first week of January.  The 4th or later would slightly better for me.  So set it up and let me know when.
